CHANGELOG.md

```diff
@@ -1,6 +1,94 @@
 # Unreleased
 
-Nothing Yet!
+This release was probably going to be several releases, but everything got finished at the same time, so here's a Mega Release!
+
+The headline features are:
+
+* New Support For Axo Releases, As An Alternative To Github Releases (Launching Soon™)
+* New Support For Generic Build Steps, Using Things Like Makefiles (Experimental)
+* Significantly Improved MSI Installer Support
+
+## Features
+
+### Axo Releases
+
+Axo Releases users can now enable builtin cargo-dist support by setting `hosting = "axodotdev"` in their
+`[workspace.metadata.dist]`.
+
+Axo Releases has a more robust pipelined model for creating and hosting a release, which more
+closely matches the actual design of cargo-dist. But since we'd only ever supported Github Releases,
+some significant internal reckoning was required.
+
+This reckoning primarily appears in the existence of the new "cargo dist host" subcommand, which
+was created to make "side-effectful networking" explicit, instead of riddling several random commands
+with various --dry-run flags.
+
+`host` takes several --steps:
+
+* create: ask Axo Releases to create hosting for the Apps we want to publish
+* upload: upload built Artifacts to the hosting that `create` made
+* release: create Releases for the hosted artifacts, making perma-urls like /v1.0.0/ live
+* announce: announce all the Releases, wiring them into "list all releases" and "latest release" endpoints
+* check: equivalent to `create` but just checks that authentication is properly setup, without side-effects
+
+The distinction between upload, release, and announce in particular lets us provide a more
+reliable/transactional release process -- we can make the hosting live, publish to package managers,
+and *then* update URLs like /latest/ once everything works, instead of racily doing it all
+at once and having to frantically hack things back to normal when something weird happens.
+It should also make it possible for us to provide features like Release/PR Previews.
+
+* impl
+    * @gankra [preparatory refactor](https://github.com/axodotdev/cargo-dist/pull/546)
+    * @gankra [create gazenot client library](https://github.com/axodotdev/gazenot)
+    * @mistydemeo [break tag parsing into "axotag" crate](https://github.com/axodotdev/cargo-dist/pull/567)
+    * @gankra [properly set announcement body for abyss](https://github.com/axodotdev/cargo-dist/pull/586)
+
+
+### Generic Builds
+
+TODO: misty writes this
```

Comment:
Here's what I wrote in my draft:

```suggestion
0.5.0 contains experimental support for building non-cargo-based projects. These can be in any language, and follow any repository layout, so long as they're accompanied by a cargo-dist manifest file that provides information on how to build and install it. For more information, consult the documentation.
```

Since it's experimental I kept it short, but I can expand if we want that at this point.